src/groovy/au/org/emii/portal/wms/GeoserverServer.groovy

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2014 IMOS
+ *
+ * The AODN/IMOS Portal is distributed under the terms of the GNU General Public License
+ *
+ */
+
+package au.org.emii.portal.wms
+
+import au.org.emii.portal.Layer
+import au.org.emii.portal.Server
+import au.org.emii.portal.proxying.ExternalRequest
+import groovy.xml.MarkupBuilder
+
+class GeoserverServer extends WmsServer {
+    def filterSource
+
+    GeoserverServer(filterSource) {
+        this.filterSource = filterSource
+    }
+
+    def getStyles(server, layer) {
+
+        return []
+    }
+
+    def getFilters(server, layer) {
+        def filters = []
+
+        try {
+            def xml = new XmlSlurper().parseText(_getFiltersXml(server, layer))
+
+            xml.filter.each { filter ->
+
+                filters.push([
+                    label: filter.label.text(),
+                    type: filter.type.text(),
+                    name: filter.name.text(),
+                    visualised: Boolean.valueOf(filter.visualised.text()),
+                    wmsStartDateName: filter.wmsStartDateName.text(),
+                    wmsEndDateName: filter.wmsEndDateName.text()
+                ])
+            }
+        }
+        catch (e) {
+            log.error "Unable to parse filters for server '${server}', layer '${layer}'", e
+        }
+
+        return filters
+    }
+
+    def _getFiltersXml(server, layer) {
+
+        switch (filterSource) {
```

Comment:
Well, I guess this answers my question from above.  But, in this case, I don't think it's a feature toggle - it's just configuration.  A feature toggle is like a light switch - on or off.